src/plugins/resolver/resolver.c

```diff
@@ -73,6 +73,9 @@ static void resolverInit (resolverHandle * p, const char * path)
 	p->tempfile = 0;
 
 	p->path = path;
+
+	p->uid = getuid ();
+	p->gid = getgid ();
```

Comment:
Yes, seems to be that those two syscalls are done useless in this initialization.

You should remember them somewhere else, so that the chown call is only issued when we are root (and chown will not fail anyway). Maybe there is even a better way to check if we are able to issue chown (root comparision seems to be inaccurate, actually we want to check the capability SYS_CHOWN, but this seems to be very unportable,..).